Data/Binary/Serialise/CBOR/Class.hs

```diff
@@ -1023,7 +1034,25 @@ instance Serialise UTCTime where
             TypeFloat32 -> utcFromReal <$> decodeFloat
             TypeFloat64 -> utcFromReal <$> decodeDouble
             _ -> fail "Expected numeric type following tag 1 (epoch offset)"
-        _ -> fail "Expected timestamp (tag 0 or 1)"
+        40 -> do
+          len <- decodeListLen
+          when (len /= 2) $ fail "Expected list of length two (UTCTime)"
+          tt <- peekTokenType
+          days <- case tt of
+            TypeUInt    -> utcFromIntegral <$> decodeWord
+            TypeUInt64  -> utcFromIntegral <$> decodeWord64
+            TypeNInt    -> utcFromIntegral <$> decodeInt
+            TypeNInt64  -> utcFromIntegral <$> decodeInt64
+            TypeInteger -> utcFromIntegral <$> decodeInteger
+          tt <- peekTokenType
+          picosecs <- case tt of
+            TypeUInt    -> utcFromIntegral <$> decodeWord
+            TypeUInt64  -> utcFromIntegral <$> decodeWord64
+            TypeNInt    -> utcFromIntegral <$> decodeInt
+            TypeNInt64  -> utcFromIntegral <$> decodeInt64
+            TypeInteger -> utcFromIntegral <$> decodeInteger
+          return (UTCTime (fromEnum days) (picosecondsToDiffTime picosecs))
+        _ -> fail "Expected timestamp (tag 0, 1 or 2)"
```

Comment:
Also, `diffTimeToPicoseconds` and `picosecondsToDiffTime` need to be imported from `Data.Time` according to Travis...